### PR TITLE
wxPython 3.0 compatibility

### LINKED ISSUE #924: Support wxPython 3.0
> _Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=888) by @yanne on 17 Nov 2011_

We explicitly select wxPython 2.8 if it is found using  wxversion.select('2.8') in roboide/**init**.py

However, wxPython 2.9 has been available for some time, so we should prefer that if it's found.

### PR DESCRIPTION
**Pull request by HelioGuilherme66**
Attempt to fix #924.
Makes possible to run RIDE with wxPython 3.0.2.0 (minimal 2.8.12.1).
COMPATIBLE WITH 2.8.12.1, 2.9.5 and 3.0.2.
Tested in Windows 7 x64 wxPython 2.8.12.1, 2.9.5 and 3.0.2 and Fedora 22 x64. Both systems tested with Python 2.7.10 and RF 2.9.3 dev. This is RIDE 1.5.
~~(The Recent Files plugin is disabled in code until fix)~~

### CODE REVIEW

**Comment by HelioGuilherme66** [comment]
The log file is OK, at least after closing RIDE:
```
more /tmp/8a7fad7f-b870-4a98-948e-b9ec0a20c958-ride.log
20151106 22:25:57.524 [INFO]: Found Robot Framework version 2.9.3.dev20151102 from '/usr/lib/python2.7/site-packages/robotframework-2.9.3.dev20151102-py2.7.egg/robot'.

20151106 22:25:57.525 [INFO]: Started RIDE 1.5a2 using python version 2.7.10 with wx version 3.0.2.0 in linux2.
```

**Comment by jplambert** [comment]
Hello,

What is the progress on this Pull Request? I'd be happy not to rely on an old version of wxpython anymore :-)

Thanks in advance.

**Comment by HelioGuilherme66** [comment]
Hello,

We have a package for you to try, at https://github.com/HelioGuilherme66/RIDE/releases/tag/v1.6a1
In Mac OS X, there are reports of success with wxPython 2.8.12.1 but crashes with 2.9 and 3.0 versions. If you are on Windows or Linux it will be OK.
(there are some problems with GUI use, for example the Keyword help in Windows is not able to detach the window)

**Comment by HelioGuilherme66** [comment]
Some known problems in wx 3.0 (mostly in Windows):
- Library import dialog, does not allow to type name of library (only by Browse button)
- Copy + Paste functions (menus or keyboard) are not working in Grid Editor
- Keywords context help is not detachable from Grid Editor, and help text vertical scroll bar not working
- Options panel for "jybot" are added to existing panel for "pybot" (still works, but we must know which one to use)

I am not seeing much activity or interest in this PR. (I know this is wanted thing since 2011, but I have the feeling people are giving up). Please manifest your interest in helping to getting RIDE modernized. Thanks.

**Comment by jerry57** [comment]
You are correct in that I think most folks have given up on this unfortunately.  4 years is a long time with little activity.  Also, not all of us have access to "test" machines to check this out.  I am going to wipe and reset a machine this morning just for looking at this if all goes okay.

**Comment by serhatbolsu** [comment]
+1 for interest, even like +10 that our customer is looking forward to it

File: src/robotide/application/pluginconnector.py [hunk 1]
```diff
@@ -20,7 +20,7 @@ def PluginFactory(application, plugin_class):
     try:
         plugin = plugin_class(application)
     except Exception:
-        return BrokenPlugin(utils.get_error_details(), plugin_class)
+        raise # return BrokenPlugin(utils.get_error_details(), plugin_class)
```
Comment (yanne) [inline src/robotide/application/pluginconnector.py]: This change should be reverted, I think.

File: src/robotide/application/pluginconnector.py [hunk 2]
```diff
@@ -65,8 +65,8 @@ class BrokenPlugin(_PluginConnector):
 
     def __init__(self, error, plugin_class):
         name = utils.name_from_class(plugin_class, 'Plugin')
-        doc = 'This plugin is disabled because it failed to load properly.\n' \
-               + 'Error: ' + error
+        doc = "This plugin is disabled because it failed to load properly.\n" \
+        % "Error: " % error
```
Comment (yanne) [inline src/robotide/application/pluginconnector.py]: Let's try to use `''.format()` as much as possible.

File: src/robotide/context/coreplugins.py [hunk 3]
```diff
@@ -23,7 +22,9 @@ def get_core_plugins():
     from robotide.log import LogPlugin
     from robotide.searchtests.searchtests import TestSearchPlugin
     from robotide.spec.specimporter import SpecImporterPlugin
+    from robotide.recentfiles import RecentFilesPlugin
 
-    return [RunAnything, RecentFilesPlugin, PreviewPlugin, SpecImporterPlugin,
-            EditorPlugin, TextEditorPlugin, KeywordSearch, LogPlugin, TestSearchPlugin]
+    # removed RecentFilesPlugin,
+    return [RunAnything, PreviewPlugin, SpecImporterPlugin,
+            EditorPlugin, TextEditorPlugin, KeywordSearch, LogPlugin, TestSearchPlugin, RecentFilesPlugin]
```
Comment (yanne) [inline src/robotide/context/coreplugins.py]: Why is the order of plugins just changed? Maybe revert all the changes to this file.

File: src/robotide/log/log.py [hunk 4]
```diff
@@ -80,10 +83,10 @@ def _create_menu(self):
 
     def _log_message(self, log_event):
         self._log.append(log_event)
-        if self._window:
-            self._window.update_log()
+        if self._panel:
+            self._panel.update_log()
         if self.log_to_console:
-            print _message_to_string(log_event)
+            print >> sys.stdout, _message_to_string(log_event)
```
Comment (yanne) [inline src/robotide/log/log.py]: The `print` statement already uses sys.stdout, so this does nothing

**Comment by yanne** [comment]
I made some general code style comments here. If you @HelioGuilherme66 can comment/change based on those, I could do an official alpha release with my own changes/fixes included.

**Comment by HelioGuilherme66** [comment]
@yanne Done the changes you suggested.

**Comment by yanne** [comment]
@HelioGuilherme66 I've picked all the relevant changes from this branch to wxPython3-compatibility branch in the main repo. This PR can be closed. Please checkout that branch for all future development. 

I'll try to make an alpha release ASAP.

**Pull request closed without merging**